File: src/main/scala/nu/tengstrand/tetrisanalyzer/piece/PieceI.scala
package nu.tengstrand.tetrisanalyzer.piece

class PieceI extends Piece {
  val number = 1.toByte
  val character = 'I'
  protected val widths = Array(4, 1)
  protected val heights = Array(1, 4)
  protected val shapes = Array(
    new PieceShape(Array(Point(0,0), Point(1,0), Point(2,0), Point(3,0))),
    new PieceShape(Array(Point(0,0), Point(0,1), Point(0,2), Point(0,3)))
  )
}

File: otherlang/src/main/java/com/github/tetrisanalyzer/piece/PieceI.java
package nu.tengstrand.tetrisanalyzer.piece;

public class PieceI extends Piece {
    @Override public byte number() { return 1; }
    @Override public char character() { return 'I'; }
    @Override protected int[] widths() { return new int[] { 4, 1 }; }
    @Override protected int[] heights() { return new int[] { 1, 4 }; }
    @Override protected PieceShape[] shapes() {
        return new PieceShape[] {
            new PieceShape(new Point(0,0), new Point(1,0), new Point(2,0), new Point(3,0)),
            new PieceShape(new Point(0,0), new Point(0,1), new Point(0,2), new Point(0,3))
        };
    }
}
